tests/integration/test_fallback.py

```diff
@@ -28,8 +28,8 @@ def test_ufunc():
     in_num = num.array([0, 1, 2, 3])
     in_np = in_num.__array__()
 
-    out_num = np.logical_and.reduce(in_num)
-    out_np = np.logical_and.reduce(in_np)
+    out_num = np.logical_or.reduce(in_num)
+    out_np = np.logical_or.reduce(in_np)
```

Comment:
In this PR, I would probably also give the logical or (and maybe xor) binary ops reduction codes, and then just remove these fallback tests

Reply:
I can add in logical_or very easily with a sum reduction code, but these are the only actual functions used for test_fallback. Would it be fine to find some other obscure, unimplemented numpy function to maintain this test?

Reply:
I see. @manopapad what do you want to do here to keep testing the fallback?

Reply:
Let's just put `logical_xor` here, and just cover `logical_and` and `logical_or` in this PR.

Also, since we're here, let's copy this block from `test_array_fallback.py`:

```
    # This test uses logical_xor.reduce because it is currently unimplemented,
    # and we want to verify a behaviour of unimplemented ufunc methods. If
    # logical_xor.reduce becomes implemented in the future, this assertion will
    # start to fail, and a new (unimplemented) ufunc method should be found to
    # replace it
    assert not logical_xor.reduce._cunumeric.implemented
```

Reply:
Based on @JosephGuman's [comment](https://github.com/nv-legate/cunumeric/pull/1123#discussion_r1487278232) below, we can't rely on `_cunumeric.implemented` for `reduce`. So let's try using `logical_and.accumulate._cunumeric.implemented` instead, since that's not implemented at all.